src/ui/checkerboard.js

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018 The boardgame.io Authors.
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Checkerboard
+ *
+ * Component that will show a configurable checker board for games like
+ * chess, checkers and others. The vertical columns of squares are labeled
+ * with letters from a to z, while the rows are labeled with numbers, starting
+ * with 1.
+ *
+ * Props:
+ *   rows    - How many rows to show up, 8 by default.
+ *   cols    - How many columns to show up, 8 by default. Maximum is 26.
+ *   onClick - On Click Callback, (row, col) of the square passed as argument.
+ *   primaryColor - Primary color, #d18b47 by default.
+ *   secondaryColor - Secondary color, #ffce9e by default.
+ *   colorMap - Object of object having cell names as key and colors as values.
+ *   Ex: { 'c5': 'red' } colors cells c5 with red.
+ *
+ * Usage:
+ *
+ * <Checkerboard>
+ *   <Token square={'c5'}>
+ *     <Knight color='dark' />
+ *   </Token>
+ * </Checkerboard>
+ */
+class Checkerboard extends React.Component {
+  static propTypes = {
+    rows: PropTypes.number,
+    cols: PropTypes.number,
+    onClick: PropTypes.func,
+    primaryColor: PropTypes.string,
+    secondaryColor: PropTypes.string,
+    colorMap: PropTypes.object,
+    style: PropTypes.object,
+    children: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.element),
+        PropTypes.element
+    ]),
+  };
+
+  static defaultProps = {
+    rows: 8,
+    cols: 8,
+    onClick: () => {},
+    primaryColor: '#d18b47',
+    secondaryColor: '#ffce9e',
+    colorMap: {},
+    style: {}
+  };
+
+  render() {
+    // Children
+    const childrenWithExtraProp = React.Children.map(this.props.children,
+      child => {
+        return React.cloneElement(child, {
+          _coordinateFn: this.algebraicCord.bind(this)
+        });
+      }
+    );
+    return (<svg viewBox={'0 0 ' + this.props.cols + ' ' + this.props.rows}
```

Comment:
Can we use a `Grid` here instead of SVG directly?

Reply:
The issue with using ```Grid``` is that this implementation of Checkerboard has its own coordinate system (implementing _coordinateFn), and ```Grid``` has another implementation for _coordinateFn. Because the coordinate system uses ```<Token/>``` props, we can pass only one or another.

I prefer to create a coordinate system using "a1" notation because it encapsulates the positioning of the elements on the screen totally on Checkerboard, and the board.js does not need to know anything about it (just the position like "c5").

Reply:
@nicolodavis let me know if you have some idea on how to make this better, im happy to change